### PR TITLE
Add tags variable to terraform

### PR DESCRIPTION
**Pull request by ralren**
[Pivotal Story](https://www.pivotaltracker.com/story/show/172481131)

This ticket adds a new variable to allow a user of the module to add their own tags to the logs s3 bucket.

This implementation is modeled after [terraform-aws-rds](https://github.com/terraform-aws-modules/terraform-aws-rds/blob/master/variables.tf#L219)

### CODE REVIEW

**Comment by ralren** [comment]
Ah, an edge case I haven't thought of. I don't know but I'll run a case where that happens.

**Comment by ralren** [comment]
Well, the tests ran fine interestingly. I don't know what it does with it though. @eeeady

File: variables.tf [hunk 1]
```diff
@@ -140,3 +140,9 @@ variable "nlb_logs_prefixes" {
   default     = ["nlb"]
   type        = list(string)
 }
+
+variable tags {
+  type        = map(string)
+  default     = {}
+  description = "A mapping of tags to assign to the logs bucket."
```
Comment (mdawn) [inline variables.tf:147]: You may have mentioned this in slack already...why did you choose a map of strings again? Is that the typical structure for tags or something?

Comment (ralren) [reply variables.tf:147]: Tags, as setup within an `aws_s3_bucket` resource, already looks like a map of strings where you have a collection of key-value pairs. It made sense to me to have the variable `tags` (the optional tags a user can specify in a module) also be a map of strings. This also allows these optional tags to be merged into the default tags without doing any transformations.

**Comment by ralren** [comment]
Writing tests!
